src/QueryBuilder/TableFilterQueryBuilder.php

```diff
@@ -0,0 +1,378 @@
+
+    public function __construct(
+        Connection $connection,
+        ?TableInfo $tableInfo,
+        ColumnConverter $columnConverter
+    ) {
+        $this->connection = $connection;
+        $this->tableInfo = $tableInfo;
+        $this->columnConverter = $columnConverter;
+    }
+
+    public function buildQueryFromCommand(
+        PreviewTableCommand $options,
+        string $schemaName
+    ): TableFilterQueryBuilderResponse {
+        $this->assertFilterCombination($options);
+
+        $query = new QueryBuilder($this->connection);
+
+        $this->processChangedConditions($options, $query);
+
+        try {
+            if ($options->getFulltextSearch() !== '') {
+                if ($this->tableInfo === null) {
+                    throw new LogicException('tableInfo variable has to be set to use fulltextSearch');
+                }
+
+                $tableInfoColumns = [];
+                /** @var TableInfo\TableColumn $column */
+                foreach ($this->tableInfo->getColumns() as $column) {
+                    // search only in STRING types
+                    if ($this->getBasetype($column->getType()) === BaseType::STRING) {
+                        $tableInfoColumns[] = $column->getName();
+                    }
+                }
+
+                $this->buildFulltextFilters(
+                    $query,
+                    $options->getFulltextSearch(),
+                    $tableInfoColumns,
+                );
+            } else {
+                $this->processWhereFilters($options->getWhereFilters(), $query);
+            }
+
+            $this->processOrderStatement($options->getOrderBy(), $query);
+        } catch (QueryException $e) {
+            throw new TableFilterQueryBuilderException(
+                $e->getMessage(),
+                $e
+            );
+        }
+
+        $this->processSelectStatement($options, $query);
+        $this->processLimitStatement($options, $query);
+        $this->processFromStatement($options, $query, $schemaName);
+
+        $sql = $query->getSQL();
+
+        /** @var string[] $types */
+        $types = $query->getParameterTypes();
+
+        return new TableFilterQueryBuilderResponse(
+            $sql,
+            $query->getParameters(),
+            $types,
+            ProtobufHelper::repeatedStringToArray($options->getColumns()),
+        );
+    }
+
+    private function assertFilterCombination(PreviewTableCommand $options): void
+    {
+        if ($options->getFulltextSearch() !== '' && $options->getWhereFilters()->count()) {
+            throw new TableFilterQueryBuilderException(
+                'Cannot use fulltextSearch and whereFilters at the same time',
+            );
+        }
+    }
+
+    private function processChangedConditions(PreviewTableCommand $options, QueryBuilder $query): void
+    {
+        if ($options->getChangeSince() !== '') {
+            $query->andWhere('"_timestamp" >= :changedSince');
```

Comment:
TD quote?

Reply:
pokud to neni promenna, tak to vetsinou piseme rovnou do SQL, tady neni uzivatelsky vstup...